packages/ui/src/header/Header.test.tsx

```diff
@@ -0,0 +1,74 @@
+// 3rd party
+import { describe, expect, it, vi } from "vitest";
+import { act, render, waitFor } from "@testing-library/react";
+// repo
+import {
+  GET_TESTID_BUTTON,
+  GET_TESTID_HEADER_NAV_ITEM,
+  GET_TESTID_LINK,
+  TESTID_HEADER,
+  TESTID_HEADER_MOBILE_NAV,
+  TESTID_HEADER_MOBILE_NAV_BUTTON,
+} from "@repo/constants/src/testids";
+// local
+import { userSetupAndRender } from "../../tests/utils";
+import Header from "./Header";
+import { URL_HOME_TITLE } from "@repo/constants/src/urls";
+
+// mock next/navigation userRouter
+const routerPush = vi.fn();
+vi.mock("next/navigation", () => ({
+  useRouter: () => ({
+    push: routerPush,
+  }),
+}));
+
+describe("Header", () => {
+  it("renders", () => {
+    const { getByTestId } = render(<Header />);
+    expect(getByTestId(TESTID_HEADER)).toBeInTheDocument();
+  });
+  // test mobile nav
+  it("mobile nav: renders and appears", async () => {
+    // Set the desired window size for mobile
+    window.innerWidth = 400;
+
+    // render the component
+    const { user, queryByTestId, getByTestId } = userSetupAndRender(<Header />);
+    expect(getByTestId(TESTID_HEADER)).toBeInTheDocument();
+
+    // get mobile nav button
+    const mobileNavButton = getByTestId(
+      GET_TESTID_BUTTON(TESTID_HEADER_MOBILE_NAV_BUTTON),
+    );
+    expect(mobileNavButton).toBeInTheDocument();
+
+    // click the mobile nav button
+    await act(async () => {
+      await user.click(mobileNavButton);
+    });
+
+    // expect mobile nav to be visible
```

Comment:
Should the mobile nav not be visible before the button is clicked?

If so, it would be helpful to check that.

Reply:
Good call. Updated!